demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,526 @@
+
+/**
+ * @brief Configurations of the initial request headers that are passed to
+ * #HTTPClient_InitializeRequestHeaders.
+ */
+static HTTPRequestInfo_t requestInfo;
+
+/**
+ * @brief Represents a response returned from an HTTP server.
+ */
+static HTTPResponse_t response;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Performs a DNS lookup on the given host name, then establishes a TCP
+ * connection to the server.
+ *
+ * @param[in] pServer Host name of server.
+ * @param[in] port Server port.
+ * @param[out] pTcpSocket Pointer to TCP socket file descriptor.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int connectToServer( const char * pServer,
+                            uint16_t port,
+                            int * pTcpSocket );
+
+/**
+ * @brief The transport send function that defines the transport interface.
+ *
+ * This is passed as the #HTTPTransportInterface.send function and used to
+ * send data over the network.
+ *
+ * @param[in] pContext User defined context (TCP socket for this demo).
+ * @param[in] pBuffer Buffer containing the bytes to send over the network stack.
+ * @param[in] bytesToSend Number of bytes to write to the network.
+ *
+ * @return Number of bytes sent if successful; otherwise negative value on error.
+ */
+static int32_t transportSend( HTTPNetworkContext_t * pContext,
+                              const void * pBuffer,
+                              size_t bytesToSend );
+
+/**
+ * @brief The transport receive function that defines the transport interface.
+ *
+ * This is passed as the #HTTPTransportInterface.recv function used for reading
+ * data received from the network.
+ *
+ * @param[in] pContext User defined context (TCP socket for this demo).
+ * @param[out] pBuffer Buffer to read network data into.
+ * @param[in] bytesToRead Number of bytes requested from the network.
+ *
+ * @return Number of bytes received if successful; otherwise negative value on error.
+ */
+static int32_t transportRecv( HTTPNetworkContext_t * pContext,
+                              void * pBuffer,
+                              size_t bytesToRecv );
+
+/**
+ * @brief Send an HTTP request based on a specified method and path, then
+ * print the response received from the server.
+ *
+ * @param[in] pTransportInterface The transport interface for making network calls.
+ * @param[in] pHost The host name of the server.
+ * @param[in] pMethod The HTTP request method.
+ * @param[in] pPath The Request-URI to the objects of interest.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int sendHttpRequest( const HTTPTransportInterface_t * pTransportInterface,
+                            const char * pHost,
+                            const char * pMethod,
+                            const char * pPath );
+
+/*-----------------------------------------------------------*/
+
+static int connectToServer( const char * pServer,
```

Comment:
This should be in common code since the MQTT library (and any others which use TCP) can use it.

Reply:
Yes that's the plan: Make a common library for connect and transport interface, then update these demos after the PRs are merged in as discussed with @leegeth.

Reply:
Agree, common code will be factored out later, right now all demos in MQTT have this function.  So for this PR we will have it in the demo.